tests/bettertransformer/test_bettertransformer_audio.py

```diff
@@ -141,3 +143,109 @@ def test_raise_train(self):
+
+            # get modules:
+            hf_modules = list(hf_model.modules())
+            bt_modules = list(bt_model.modules())
+
+            # Assert that the modules are the same
+            self.assertEqual(len(hf_modules), len(bt_modules))
+            for hf_module, bt_module in zip(hf_modules, bt_modules):
+                self.assertEqual(type(hf_module), type(bt_module))
+                # check the modules have the same methods
+                self.assertEqual(dir(hf_module), dir(bt_module))
+
+                # check the modules have the same attributes
+                hf_module_attributes = [attr for attr in dir(hf_module) if not attr.startswith("_")]
+                bt_module_attributes = [attr for attr in dir(bt_module) if not attr.startswith("_")]
+
+                self.assertEqual(hf_module_attributes, bt_module_attributes)
+
+    @parameterized.expand([(False,)])
+    def test_save_load_invertible(self, keep_original_model=False):
+        r"""
+        Since `Wav2vec2` does not support `deepcopy` we cannot test the transformation
+        with `keep_original_model=True` for this model.
+        """
+        for model in self.all_models_to_test:
+            with tempfile.TemporaryDirectory() as tmpdirname:
+                hf_model = AutoModel.from_pretrained(model).eval()
+                bt_model = BetterTransformer.transform(hf_model, keep_original_model=keep_original_model)
+
+                bt_model = BetterTransformer.reverse(bt_model)
+                # check if no parameter is on the `meta` device
+
+                for name, param in bt_model.named_parameters():
+                    self.assertFalse(param.device.type == "meta", f"Parameter {name} is on the meta device.")
+
+                bt_model.save_pretrained(tmpdirname)
+
+                bt_model_from_load = AutoModel.from_pretrained(tmpdirname)
+
+                # check if the state dict is the same
+                # first check if the keys are the same
+                self.assertEqual(
+                    set(bt_model.state_dict().keys()),
+                    set(bt_model_from_load.state_dict().keys()),
+                )
+                # check also with HF model
+                self.assertEqual(
+                    set(hf_model.state_dict().keys()),
+                    set(bt_model_from_load.state_dict().keys()),
+                )
+
+                for key in bt_model.state_dict().keys():
+                    self.assertTrue(
+                        torch.allclose(
+                            bt_model.state_dict()[key],
+                            bt_model_from_load.state_dict()[key],
+                        )
+                    )
+
+                    self.assertTrue(
+                        torch.allclose(
+                            hf_model.state_dict()[key],
+                            bt_model_from_load.state_dict()[key],
+                        )
+                    )
```

Comment:
same comment, fuzzy think here because `keep_original_model = False` (we don't test against the original model, are we? It sounds like there are in place modifications)

Reply:
fixed in [34363bf](https://github.com/huggingface/optimum/pull/676/commits/34363bf8f9ad9c29ea0cdfae4580062afebf907d)